### PR TITLE
Only log error if replies is not successful and not 404

### LINKED ISSUE #43: 🐛 Bug: Don't call console methods if the repo doesn't have a .github/replies.yml
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/refined-saved-replies/issues?q=is%3Aissue+label%3A%22type%3A+bug%22) and found none that matched my issue.

### Expected

When a user with this extension visits a repository that doesn't have a `.github/replies.yml`, this extension shouldn't call any console methods - least of all `console.error`. It just means the repo isn't set up for refined saved replies.

### Actual

This extension's content script explicitly logs:

```plaintext
Invalid saved replies: {404: 'Not Found'}
```

We should not log in that case.

### Additional Info

#5 tracks using an API that doesn't give a `console.error` for a 404 response. That's out of scope of this issue.

### PR DESCRIPTION
**Pull request by estevezluis**
## PR Checklist

- [x] Addresses an existing open issue: fixes #43 
- [x] That issue was marked as [accepting prs](https://github.com/JoshuaKGoldberg/refined-saved-replies/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/refined-saved-replies/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

removes "Invalid saved replies" error log

### CODE REVIEW

File: src/content-script.ts [hunk 1]
```diff
@@ -41,7 +41,6 @@ import { isBodyWithReplies, isRepositoryDetails } from "./validations";
 	// 5. Parse the replies body as yml
 	const repliesConfiguration = yaml.load(repliesBody);
 	if (!isBodyWithReplies(repliesConfiguration)) {
-		console.error("Invalid saved replies:", repliesConfiguration);
```
Comment (JoshuaKGoldberg) [inline src/content-script.ts:44]: [Bug] Hmm, what if the response is like a 400 or 500? I think we'll still want to `console.error` in that case.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for starting on this @estevezluis! I think we'll need a bit more logic though. ✨

File: src/content-script.ts (hunk 1 above)
Comment (estevezluis) [reply src/content-script.ts]: Ahh yes, thank you for pointing that out! PR updated!

Comment (estevezluis) [reply src/content-script.ts]: @JoshuaKGoldberg Do we want to log failed fetch for the other fetch calls? If so, I can open an issue for those as well.

File: src/content-script.ts [hunk 2]
```diff
@@ -36,6 +36,14 @@ import { isBodyWithReplies, isRepositoryDetails } from "./validations";
 	// 4. Fetch the repository's .github/replies.yml
 	const repliesUrl = `https://raw.githubusercontent.com/${userOrOrganization}/${repository}/${defaultBranch}/.github/replies.yml`;
 	const repliesResponse = await fetch(repliesUrl);
+
+	if (!repliesResponse.ok) {
+		if (repliesResponse.status !== 404) {
+			console.error("Fetch replies:", repliesResponse.statusText);
```
Comment (JoshuaKGoldberg) [inline src/content-script.ts:42]: [Nitpicking] Just making this a bit more understandable to layfolk:

```suggestion
			console.error("Non-ok response fetching replies:", repliesResponse.statusText);
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, thanks!

**Pull request merged**